src/main/java/pl/put/bulidinginfo/models/RoomCompound.java

```diff
@@ -51,13 +53,12 @@ public List<Location> getLocationsWithHigherHeatingRate(Float thresholdRate) {
         for (Location floor : getLocations()) {
             if (floor instanceof RoomCompound) {
                 for (Location room : ((RoomCompound) floor).getLocations()) {
-                    if (room.getHeatingPerCubeMeter() > thresholdRate) {
-                        filteredLocations.add(room);
-                    }
+                    filteredLocations.addAll(room.getLocationsWithHigherHeatingRate(thresholdRate));
                 }
             }
```

Comment:
Teraz możemy zamiast sprawdzać czy to pietro od razu wywołać bez fora po pokojach filteredLocations.addAll(room.getLocationsWithHigherHeatingRate(thresholdRate)); 
jest to uniwersalniejsze i czytelniejsze